### DISCUSSION TITLE
Welcome to FMT-Firmware Discussions!

### DISCUSSION THREAD

**Discussion by JcZou** [Firmament-Autopilot/FMT-Firmware#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by meshaolong** [comment]
你好，Jiachi！感谢你开源分享基于RT-Thread的无人机固件，也拜读了你的论文《Robust Active Disturbance Rejection Control
Scheme for Quadrotor UAVs: Experimental Prototyping》。我们组是做水下机器人方向，想学习下您这个开源固件，如果您这个合适，也希望把FMT-Firmware扩展到水下领域。不知道您有没有更详细的关于您开发工程的代码开发方面的介绍材料分享，希望能更多了解。谢谢分享，我的邮箱是362356496@qq.com。

**Comment by JcZou** [comment]
你好，很高兴你对 FMT 感兴趣！目前项目的资料可以参考[项目文档](https://firmament-autopilot.github.io/FMT-DOCS/#/content_ch/)（still work in progress）。里面包含了 FMT 的基本使用方法和一些核心功能介绍。
我建议你下载 FMT-Firmware 和 FMT-Model 的代码，然后根据文档步骤做一下 SIH, MIL 和 Open-loop 的仿真，这样你可以对 FMT 有个大致了解。如果要更进一步的话，代码是最好的学习资料。
如果有其它任何问题也欢迎来问我~

**Comment by zhongzhw** [comment]
Hi, FMT is a pretty good projet, and we are looking for a research plantform to realizing our UAV flight control algorithm. It seems FMT is a good one, I have some question about this firmware
1. STM32H7 mcu has been tested or not
2. what should I do if I want to transplant this firmware to other stm32 chip, like stm32H7(the peripherals init can be generated with cubeMX?)
3. IMU and mag calibration program also seems be auto generated, but I didn't see the matlab code
Thank you for your contribute, and wating for your reply!

**Reply by JcZou** [reply]
Hi, thanks for your interest ;-) FMT is definitely a good platform for research with your own control algorithm.

For your questions:
1. Support with STM32H7 is on going. We are working on porting FMT to HEX orange cube.
2. If you would like to port FMT to STM32H7 by yourself, what you should do is to create a folder in target/ and implement all files in it, you can refer to target/pixhawk/fmu-v5 folder. Yes, the peripheral drivers can be generated with cubeMX and it's recommended way.
3. Exactly, sensor calibration algorithm is designed in MATLAB and code is generated. It is well designed and tested, so you can just simply  use it. The original matlab code is not ready to become public yet.

**Comment by w2zm** [comment]
您好，我对FMT的模型设计有一点疑问。我在plant的模型中没有看到传统的无人机数学模型，请问如何修改运动学和动力模型呢，还是只能接受现有的传感器输入进行处理。

**Comment by JcZou** [comment]
Plant中有无人机的数学模型（动力学和作动器模型）

**Comment by w2zm** [comment]
您好，请问一下FMT有没有交流群，关于模型的问题还想请教




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年6月25日(星期六) 晚上10:33
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Firmament-Autopilot/FMT-Firmware] Welcome to FMT-Firmware Discussions! (Discussion #6)





 
Plant中有无人机的数学模型（动力学和作动器模型）

**Reply by JcZou** [reply]
可以加QQ群：459133925
